reconcile/utils/openshift_resource.py

```diff
@@ -431,6 +433,15 @@ def calculate_sha256sum(body):
         return m.hexdigest()
 
 
+def checkExpirationDate(role):
+    exp_date = datetime.datetime \
+        .strptime(role['expirationDate'], '%Y-%m-%d').date()
+    if exp_date < datetime.date.today():
+        return True
+    elif (exp_date - datetime.date.today()).days > EXPIRATION_MAX:
+        return False
+
```

Comment:
you don't cover the the situation where the expiration date is in less than 90d from now, which results in a return value of None.

if the intent of this function is to detect expiration dates, that are more than 90d in the future, you could simply like this

return (exp_date - datetime.date.today()).days <= EXPIRATION_MAX

but please verify!

Reply:
As an alternative (pick the one you prefer), all the callers in this PR are raising an exception in case the role has expired which means you can make this method act like an assertion. That is:

```python
if (exp_date - datetime.date.today()).days > EXPIRATION_MAX:
    raise ...
```
and you save yourself a few ifs in the callers.